client/utils/events.ts

```diff
@@ -445,6 +446,38 @@ const getDateStringForEvent = (event, dateOnly = false, useLocal = true, withTim
         }
     }
 
+    const isFullDay = event?.dates?.all_day;
+    const noEndTime = event?.dates?.no_end_time;
+
+    const multiDay = !isEventSameDay(start, end);
+
+    if (isFullDay && !multiDay) {
+        return datesToShow = start.format(dateFormat);
+    } else if (noEndTime && !multiDay) {
+        if (withTimezone) {
+            if (!useLocal && (tz || tz == null)) {
```

Comment:
not sure about that `tz` condition, it can be set or null, so what are we actually checking there?

Reply:
In (noEndTime && !multiDay) scenario when there is utc `tz `we are getting null so for that we have to add this extra condition. For other time zones it is working properly.

Reply:
my point is that `(tz || tz == null)` should be pretty much always true, or is there a case when it's not?

Reply:
It is `true` when `withTimezone` is `true`.If flag `withTimezone` is true we should have `tz `but in `utc` scenario we are getting null that's why we are assigning `utc` only if `withTimezone` is true and then `tz` is `null`

Reply:
I see, but then this condition `(tz || tz == null)` is not needed, is it?

Reply:
IMO it is needed when there is `UTC` timezone we have to show it in preview but we are getting null. please let me know if I am missing something.

Reply:
I think that:
```
if (!useLocal) {
```
would do because that `(tz || tz == null)` is always true in
```
if (!useLocal && (tz || tz == null)) {
```